### PR TITLE
Fix Rust beta compiler errors (1.77)

### PR DESCRIPTION
**Pull request by jimmygchen**
## Issue Addressed

Fix compilation/lint errors when compiling lighthouse using 1.77 beta compiler.

The fixes for `unused code` aren't ideal, but I can't think of a better way rn (related issue [here](https://github.com/rust-lang/rust/issues/119645)):

```rust
#[derive(Debug)]
// We don't use the inner values directly, but they're used in the Debug impl.
#[allow(dead_code)]
enum AttestationPerformanceError {
    BlockReplay(BlockReplayError),
    BeaconState(BeaconStateError),
    ParticipationCache(ParticipationCacheError),
    UnableToFindValidator(usize),
}
```

### CODE REVIEW

File: beacon_node/execution_layer/src/test_utils/mod.rs [hunk 1]
```diff
@@ -487,6 +487,7 @@ pub struct StaticNewPayloadResponse {
     should_import: bool,
 }
 #[derive(Debug)]
+// We don't use the string value directly, but it's used in the Debug impl which is required by `warp::reject::Reject`.
```
Comment (michaelsproul) [inline beacon_node/execution_layer/src/test_utils/mod.rs]: don't think we need this comment because of the `e.0` below

Comment (jimmygchen) [reply beacon_node/execution_layer/src/test_utils/mod.rs]: oh yeah I removed the `allow` attribute but forgot to remove the comment. I'll fix!

Comment (jimmygchen) [reply beacon_node/execution_layer/src/test_utils/mod.rs]: https://github.com/sigp/lighthouse/pull/5180/commits/3b1da953e9d61d674348e30fd3838e4ffc9d19bf

File: beacon_node/execution_layer/src/test_utils/mod.rs [hunk 2]
```diff
@@ -600,7 +600,7 @@ async fn handle_rejection(err: Rejection) -> Result<impl warp::Reply, Infallible
     let message;
 
     if let Some(e) = err.find::<AuthError>() {
```
Comment (jxs) [inline beacon_node/execution_layer/src/test_utils/mod.rs]: nit:
```suggestion
    if let Some(AuthError(e)) = err.find::<AuthError>() {
        message = format!("Authorization error: {}", e);
```

I couldn't include the delete line in the suggestion so had to make it to this one sorry

File: beacon_node/lighthouse_network/tests/common.rs [hunk 3]
```diff
@@ -42,7 +42,12 @@ pub fn fork_context(fork_name: ForkName) -> ForkContext {
     ForkContext::new::<E>(current_slot, Hash256::zero(), &chain_spec)
 }
 
-pub struct Libp2pInstance(LibP2PService<ReqId, E>, exit_future::Signal);
+pub struct Libp2pInstance(
+    LibP2PService<ReqId, E>,
+    #[allow(dead_code)]
+    // This field is managed for lifetime purposes may not be used directly, hence the `#[allow(dead_code)]` attribute.
+    exit_future::Signal,
+);
```
Comment (jxs) [inline beacon_node/lighthouse_network/tests/common.rs:50]: NB: submitted #5183, we don't need `exit_future`

**Review by jxs — APPROVED** [review]
LGTM Jimmy

File: beacon_node/lighthouse_network/tests/common.rs (hunk 3 above)
Comment (realbigsean) [reply beacon_node/lighthouse_network/tests/common.rs:50]: I'm thinking we merge this one first to get CI working and update this in #5183

**Comment by realbigsean** [comment]
@mergifyio queue

**Pull request merged**